in_toto/model.go

```diff
@@ -150,6 +190,67 @@ func (l *Layout) InspectAsInterfaceSlice() []interface{} {
 	return inspectionsI
 }
 
+func (l *Layout) validate() {
```

Comment:
This is not a stab at lolnogenerics, but can we generalize this function?

Reply:
Also, it feels like these functions should probably be throwing an error

Reply:
Yep, I need to update them to return errors.

Reply:
+1 generalization
+1 returning errors